### PR TITLE
Finish off test coverage

### PR DESCRIPTION
**Pull request by Hang-Lei-NOAA**
Status: Open.
issue #164

COVERAGE SUMMARY FOR NEW TESTS
================================
test_bacio_comprehensive.F90:
- Line 420-421 (baciof.F90): Zero-length read (KA = 0, RETURN)
- Line 479-486 (baciof.F90): Buffered read continuation
- Line 547-548, 551-552, 617-618, 621 (baciof.F90): Invalid LU handling
- Line 83, 89, 262, 266-270 (bafrio.F90): Endian-specific paths
- Line 36-41 (chk_endianc.F90): Endian detection (big/little/mixed)
- 
test_bafrio_errors.F90:
- Line 101 (bafrio.F90): LX=-1 (I/O error reading record)
- Line 184 (bafrio.F90): KA=LX (record length handling)
- Line 186 (bafrio.F90): KA=-3 (request longer than record)
- Line 190 (bafrio.F90): KA=-1 (I/O error)
- Line 278, 282, 286 (bafrio.F90): Write error conditions (KA=-1)

test_char0_handling.c:
- Line 441 (bacio.c): CHAR(0) parameter handling
- Filename edge cases
- Null character in filename scenarios

### CODE REVIEW

File: tests/test_bacio_comprehensive.F90 [hunk 1]
```diff
@@ -0,0 +1,419 @@
+! Comprehensive test program for missing coverage lines
```
Comment (AlexanderRichert-NOAA) [inline tests/test_bacio_comprehensive.F90]: Two thoughts: 1) Would it make sense to incorporate these tests into the existing files, or is the functionality different enough that it wouldn't really help to consolidate code? 2) I would suggest calling this something other than "comprehensive." It makes sense in the context of these changes but might be a little confusing to someone a year from now looking at the tests for the first time, as the name would I think suggest that it tests the whole library.

Comment (Hang-Lei-NOAA) [reply tests/test_bacio_comprehensive.F90]: Yes, I will updated it. I try to incorporate it to the existing code. Some are still there in the test_bacio_c.c. But other caused more complicated setting for existing code, so I make them independent. I will rename them.

Comment (AlexanderRichert-NOAA) [reply tests/test_bacio_comprehensive.F90]: I think the CI was all disabled because it's been so long since we've run it :) Can you trigger the CI?

**Comment by Hang-Lei-NOAA** [comment]
combined into pull request #166
